apollo-router/src/spec/schema.rs

```diff
@@ -237,22 +238,45 @@ impl Schema {
         self.diagnostics.is_some()
     }
 
-    pub(crate) fn has_spec(&self, url: &str) -> bool {
+    pub(crate) fn has_spec(&self, base_url: &str, min_version: &str) -> bool {
         self.definitions
             .schema_definition
             .directives
             .iter()
             .filter(|dir| dir.name.as_str() == "link")
             .any(|link| {
-                link.argument_by_name("url")
+                if let Some(url_in_link) = link
+                    .argument_by_name("url")
                     .and_then(|value| value.as_str())
-                    == Some(url)
+                {
+                    let Some((base_url_in_link, version_in_link)) = url_in_link.rsplit_once("/v")
+                    else {
+                        return false;
+                    };
+
+                    let Some(version_in_url) =
+                        Version::parse(format!("{}.0", version_in_link).as_str()).ok()
+                    else {
+                        return false;
+                    };
+
+                    let Some(min_version) =
+                        Version::parse(format!("{}.0", min_version).as_str()).ok()
+                    else {
+                        return false;
+                    };
+
+                    base_url_in_link == base_url && version_in_url >= min_version
+                } else {
+                    false
+                }
             })
     }
 
     pub(crate) fn directive_name(
         schema: &apollo_compiler::schema::Schema,
-        url: &str,
+        base_url: &str,
+        min_version: &str,
```

Comment:
why only `min_version`? If we change a directive to add a field and expect a meaningful behaviour change from it, a router that does not know how to interpret that directive should reject it

Reply:
Let me know what you think here
[9803b4b](https://github.com/apollographql/router/pull/4521/commits/9803b4bc9ad07b7783f362b66d148fd4500bb567)

Reply:
Is this implementation meant to align with the [Versioning](https://specs.apollo.dev/core/v0.2/#sec-Versioning) section of the Core specification?  (I started an internal conversation about [`link`](https://specs.apollo.dev/link/v1.0/#sel-EADHNCAACKBrkP) vs `core`, but for the purposes of this PR, I am mostly just questioning whether — for review purposes — the _intent_ is for this to match that implementation as defined in the `core` spec, irregardless of the overall spec-status itself.)